src/stream/xml_reader.rs

```diff
@@ -74,7 +74,6 @@ impl From<XmlReaderError> for ErrorKind {
                 Err(err) => ErrorKind::Io(std::io::Error::from(err.kind())),
             },
             XmlReaderError::Syntax(_) => ErrorKind::UnexpectedEof,
-            XmlReaderError::IllFormed(_) => ErrorKind::InvalidXmlSyntax,
             XmlReaderError::Encoding(_) => ErrorKind::InvalidXmlUtf8,
             _ => ErrorKind::InvalidXmlSyntax,
```

Comment:
Could you make this exhaustive, it would be good to ensure we handle all new error cases.